File: permut2SPOJ.java
import java.io.*;

class permut2SPOJ {
        public static void main(String args[]) throws IOException
        {

        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

	int t;
	int cnt=0;

	int[] arr = new int[10000];
	
	while(true)
	{
//        while((t=Integer.parseInt(in.readLine()))!=0)
//	{

	t = Integer.parseInt(in.readLine());

	if(t==0)
	{
	break;
	}
	else
	{
//	System.out.println(t+"/////");
//	int[] arr = new int[t];

	String str = in.readLine();

	for(int i=0; i<t; i++)
	{
//	String str = in.readLine();
//	System.out.println(arr[i]);
	arr[i] = Integer.parseInt(str.split(" ")[i]);
//   System.out.println(arr[i]);

	}
int cont=0;
	for(int i=0; i<t; i++)
	{	
	if(arr[arr[i]-1] == i+1)
	{
	cont++;
	}
	}
	
	if(cont==t)
	{
	System.out.println("ambiguous");
	}
	else
	{
	System.out.println("not ambiguous");
	}
	}
	

	}
}
}	
		

File: permut2SPOJ.c
/*#include<stdio.h>

int main()
	{
	
int n,i;
int a[10000];

	do{
		
	int i=0,j=0;

	scanf("%d",&n);
	
	if(n==0) 
	{
	return 0;
	}

	for(i=0; i<n; i++)
	{
	scanf("%d",&a[i]);
	}
	
	int cnt=0;

	for(i=0; i<n; i++)
	{
	if(a[a[i]-1]==i+1)
	{
	cnt++;
	}
	}
	
	if(cnt==n)
	{
	printf("ambiguous\n");
	}
	else
	{
	printf("not ambiguous\n");
	}
	}while(n!=0);

return 0;

int cnt;

while(scanf("%d", &n)==1 && n)
	{
		for(i=0; i<n; i++) scanf("%d", &a[i]);
		for(i=0, cnt=0; i<n; i++)
			if(a[a[i]-1]==i+1)
				cnt++;
		if(cnt==n) printf("ambiguous\n");
		else printf("not ambiguous\n");
	}
	return 0;

	}

*/


#include <stdio.h>

#define MAX 100001

int main()
{
	int a[MAX], n, i, cnt;
	while(scanf("%d", &n)==1 && n)
	{
		for(i=0; i<n; i++) scanf("%d", &a[i]);
		for(i=0, cnt=0; i<n; i++)
			if(a[a[i]-1]==i+1)
				cnt++;
		if(cnt==n) printf("ambiguous\n");
		else printf("not ambiguous\n");
	}
	return 0;
}
